src/ECharts/Event.purs

```diff
@@ -37,3 +39,43 @@ listenAll chart cb =
 
   toEChartsEvent ∷ ∀ ω. Tuple String ω → EChartsEvent
   toEChartsEvent = unsafeCoerce
+
+foreign import dispatchAction_
+  ∷ ∀ e action
+  . action
+  → Chart
+  → Eff ( echarts ∷ ECHARTS |e ) Unit
+
+dispatch
+  ∷ ∀ e m
+  . MonadEff ( echarts ∷ ECHARTS | e ) m
+  ⇒ EChartsEvent
+  → Chart
+  → m Unit
+dispatch vaction chart =
+  liftEff $ dispatchAction_ action chart
+  where
+  variantPair ∷ Tuple String {}
+  variantPair = unsafeCoerce vaction
+
+  actionType ∷ String
+  actionType = case fst variantPair of
+    "legendselectchanged" → "legendToggleSelect"
+    "legendselected" → "legendSelect"
+    "legendunselected" → "legendUnSelect"
+    "datazoom" → "dataZoom"
+    "datarangeselected" → "selectDataRange"
+    "timelinechanged" → "timelineChange"
+    "timelineplaychanged" → "timelinePlayChange"
+    "pieselectchanged" → "pieToggleSelect"
+    "pieselected" → "pieSelect"
+    "pieunselected" → "pieUnSelect"
+    "mapselectchanged" → "mapToggleSelect"
+    "mapselected" → "mapSelect"
+    "mapunselected" → "mapUnSelect"
+    "focusnodeadjacency" → "focusNodeAdjacency"
+    "unfocusnodeadjacency" → "unfocusNodeAdjacency"
+    s → s
+
+  action ∷ ∀ ω. { "type" ∷ String | ω }

```

Comment:
do we need qoutes here?

Reply:
Yep. At least w/o it my editor is going crazy